flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecSink.java

```diff
@@ -75,7 +75,7 @@ public StreamExecSink(
             String description) {
         super(
                 tableSinkSpec,
-                tableSinkSpec.getTableSink().getChangelogMode(inputChangelogMode),
+                inputChangelogMode,
```

Comment:
this is clearly a bug, the `inputChangelogMode` is only a suggestion for the sink. in the end the sink decides about the final mode.

Reply:
I don't follow; it still does, this is now done in `CommonExecSink#applyKeyBy` which calls `tableSink.getChangelogMode(inputChangelogMode)`?